packetbeat/tests/system/test_0060_filtering.py

```diff
@@ -0,0 +1,125 @@
+            include_fields=None, #export all fields
+            filter_enabled=True,
+        )
+
+        self.run_packetbeat(pcap="http_minitwit.pcap",
+                            debug_selectors=["http", "httpdetailed"])
+        objs = self.read_output()
+
+        assert len(objs) == 3
+        assert all([o["type"] == "http" for o in objs])
+
+        assert objs[0]["status"] == "OK"
+        assert objs[1]["status"] == "OK"
+        assert objs[2]["status"] == "Error"
+
+        print objs[0]
+
+        assert "http.request_headers" in objs[0]
+        assert "http.response_headers" in objs[0]
+
+        # generic filtering fails to delete the htp.response_headers.transfer-encoding
+        assert "transfer-encoding" not in objs[0]["http.response_headers"]
+
+
+    def test_drop_unknown_field(self):
+
+        self.render_config_template(
+            http_send_all_headers=True,
+            drop_fields=["http.response_headers.transfer-encoding-test"],
+            include_fields=None, #export all fields
+            filter_enabled=True,
+        )
+
+        self.run_packetbeat(pcap="http_minitwit.pcap",
+                            debug_selectors=["http", "httpdetailed"])
+        objs = self.read_output()
+
+        assert len(objs) == 3
+        assert all([o["type"] == "http" for o in objs])
+
+        assert objs[0]["status"] == "OK"
+        assert objs[1]["status"] == "OK"
+        assert objs[2]["status"] == "Error"
+
+        print objs[0]
+
+        assert "http.request_headers" in objs[0]
+        assert "http.response_headers" in objs[0]
+
+        # generic filtering fails to delete the htp.response_headers.transfer-encoding
+        assert "transfer-encoding" in objs[0]["http.response_headers"]
+
+    def test_include_empty_list(self):
+
+
+        self.render_config_template(
+            http_send_all_headers=True,
+            include_fields=[], # export only manadatory fields
+            filter_enabled=True,
+        )
+
+        self.run_packetbeat(pcap="http_minitwit.pcap",
+                            debug_selectors=["http", "httpdetailed"])
+        objs = self.read_output()
+
+        assert len(objs) == 3
+        assert "status" in objs[0]
+        assert "type" in objs[0]
+        assert "http.request_headers" not in objs[0]
+        assert "http.response_headers" not in objs[0]
+
+    def test_drop_no_fields(self):
+
+        self.render_config_template(
+	        http_send_all_headers=True,
+            drop_fields=[],
+            include_fields=None, #export all fields
+            filter_enabled=True,
+        )
+
+        self.run_packetbeat(pcap="http_minitwit.pcap",
+                            debug_selectors=["http", "httpdetailed"])
+        objs = self.read_output()
+
+        assert len(objs) == 3
+        assert all([o["type"] == "http" for o in objs])
+
+        assert objs[0]["status"] == "OK"
+        assert objs[1]["status"] == "OK"
+        assert objs[2]["status"] == "Error"
```

Comment:
would be good to add some `include_fields` tests. Right now I only see all in or all out tests for `include_fields`.